sdk/storage/Azure.Storage.DataMovement.Files.Shares/src/ShareFileStorageResourceOptions.cs

```diff
@@ -101,8 +94,11 @@ public class ShareFileStorageResourceOptions
 
         /// <summary>
         /// The key of the file permission.
+        ///
+        /// By default the permission key will not be preserved from the source Share to the destination Share.
         /// </summary>
-        public string FilePermissionKey { get; set; }
+        public DataTransferProperty FilePermissions { get; set; }
+        internal string _destinationPermissionKey;
```

Comment:
I added the destination permission key saved internally on the options bag here so that way when we deserialize from the checkpointer file to the options bag, we have a place to set it here.

I am open to suggestions to other places we could put it, however this seems the most fitting in terms when we create the options bag from the checkpointer.

Reply:
Remove this

Reply:
Update PathScanner to pass permission key via the Resource Properties dictionary bag.

Update ShareFIleStorageResource.GetProperties, to merge Resource Properties when fetching it from the service upon new GetProperties request

Reply:
Done